novawallet/Common/ViewController/CardLayoutPresentationController/Transitions/CardPresentingTransition.swift

```diff
@@ -0,0 +1,152 @@
+    }
+}
+
+// MARK: Private
+
+private extension CardPresentingTransition {
+    func createSourceViewTransform(_ sourceView: UIView) -> CGAffineTransform {
+        let widthDelta: CGFloat = UIConstants.horizontalInset * 2
+        let scale = (sourceView.bounds.width - widthDelta) / sourceView.bounds.width
+        let heightDiffAfterScale = (sourceView.bounds.height - (sourceView.bounds.height * scale)) / 2
+        let yOffset = sourceView.safeAreaInsets.top - heightDiffAfterScale
+
+        let sourceTransform = CGAffineTransform.identity
+        let sourceScaleTransform = CGAffineTransform(
+            scaleX: scale,
+            y: scale
+        )
+        let sourceTranslateTransform = CGAffineTransform(
+            translationX: .zero,
+            y: yOffset
+        )
+
+        return sourceTransform
+            .concatenating(sourceScaleTransform)
+            .concatenating(sourceTranslateTransform)
+    }
+
+    func createDestinationViewTransform(_ finalFrame: CGRect) -> CGAffineTransform {
+        CGAffineTransform(
+            translationX: 0,
+            y: finalFrame.minY
+        )
+    }
+
+    func createDimmingView(for view: UIView) -> UIView {
+        let dimmingView = UIView(frame: view.bounds)
+        let dimmingColor = UIColor.white.withAlphaComponent(0.1)
+
+        dimmingView.backgroundColor = dimmingColor
+        dimmingView.clipsToBounds = true
+        dimmingView.tag = dimmingViewTag
+
+        return dimmingView
+    }
+
+    func calculateFinalDestinationFrame(using sourceView: UIView) -> CGRect {
+        let finalOrigin = sourceView.bounds.offsetBy(
+            dx: 0,
+            dy: CardLayoutPresentationController.topOffset()
+        ).origin
+
+        let finalFrameSize = CGSize(
+            width: sourceView.bounds.width,
+            height: sourceView.bounds.height - CardLayoutPresentationController.topOffset()
+        )
+
+        let finalDestinationViewFrame = CGRect(
+            origin: finalOrigin,
+            size: finalFrameSize
+        )
+
+        return finalDestinationViewFrame
+    }
+}
+
+// MARK: UIViewControllerAnimatedTransitioning
+
+extension CardPresentingTransition: UIViewControllerAnimatedTransitioning {
+    func transitionDuration(using _: UIViewControllerContextTransitioning?) -> TimeInterval {
+        transitionDuration
+    }
+
+    func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
+        guard
+            let sourceController = transitionContext.viewController(forKey: .from),
+            let destinationController = transitionContext.viewController(forKey: .to)
+        else { return }
+
+        let coveredContextView: UIView
+
+        if let tabBarController = sourceController as? UITabBarController {
+            let navController = tabBarController.selectedViewController as? UINavigationController
+            coveredContextView = navController?.topViewController?.view ?? tabBarController.view
```

Comment:
Why we want to get child controller of the tab bar controller? Why not always take tabBarController?